File: MCL_by_python/WeighingFactors.py
global ZHit,ZShort,ZMax,ZRand
global m_ZHitRaw, m_ZShortRaw,m_ZMaxRaw,m_ZRandRaw


def getZHitRaw():
    global m_ZHitRaw
    return m_ZHitRaw


def getZShortRaw():
    global m_ZShortRaw
    return m_ZShortRaw

def getZMaxRaw():
    global m_ZMaxRaw
    return m_ZMaxRaw


def getZRandRaw():
    global m_ZRandRaw
    return m_ZRandRaw

def Normalize():
    global ZHit, ZShort, ZMax, ZRand

    total = getZHitRaw() + getZShortRaw() + getZMaxRaw() + getZRandRaw()

    ZHit = getZHitRaw() / total
    ZShort = getZShortRaw() / total
    ZMax = getZMaxRaw() / total
    ZRand = getZRandRaw() / total


def WeighingFactors(zHit, zShort, zMax, zRand):
    global m_ZHitRaw, m_ZShortRaw, m_ZMaxRaw, m_ZRandRaw
    m_ZHitRaw = zHit
    m_ZShortRaw = zShort
    m_ZMaxRaw = zMax
    m_ZRandRaw = zRand
    Normalize()

File: MCL_by_Cpp/MCL_C++/WeighingFactors.cpp
#include "stdafx.h"
#include <iostream>
#include "WeighingFactors.h"
#include "BeamModel.h"

double ZHitRaw=0, ZShortRaw=0, ZMaxRaw=0, ZRandRaw=0;


void Normalize()
{
   
    double total = ZHitRaw + ZShortRaw + ZMaxRaw + ZRandRaw;

    ZHit = ZHitRaw / total;
    ZShort = ZShortRaw / total;
    ZMax = ZMaxRaw / total;
    ZRand = ZRandRaw / total;
}

void WeighingFactor(double zHit, double zShort, double zMax, double zRand)
{
    ZHitRaw = zHit;
    ZShortRaw = zShort;
    ZMaxRaw = zMax;
    ZRandRaw = zRand;
    Normalize();
}